libV2/tasks/BaseTask.js

```diff
@@ -0,0 +1,82 @@
+const cron = require('node-cron');
+const cronparser = require('cron-parser');
+
+const { client: cacheClient } = require('../cache');
+const Process = require('../process');
+const { LoggerContext } = require('../utils');
+const Warp10Client = require('../warp10');
+
+const logger = new LoggerContext({
+    module: 'BaseTask',
+});
+
+class Now {}
+
+class BaseTask extends Process {
+    constructor() {
+        super();
+        this._cache = cacheClient;
+        this._warp10 = new Warp10Client();
+        this._scheduler = null;
+        this._defaultSchedule = Now;
+    }
+
+    async _setup() {
+        this._program
+            .option('-n, --now', 'Execute the task immediately and then exit. Overrides --schedule.')
+            .option(
+                '-s, --schedule <crontab>',
+                'Execute task using this crontab.  Overrides configured schedule',
+                value => {
+                    cronparser.parseExpression(value);
+                    return value;
+                },
+            );
+    }
+
+    get schedule() {
+        if (this._program.now) return Now;
```

Comment:
I'd recommend always using braces around `if`, `for` etc. and put the single instruction on the next line, because it makes the instruction more visible, and could avoid any ambiguity with `if/else` constructs that could be nested. (I think at least in some projects the linter enforces this).